regolith/helpers/u_contacthelper.py

```diff
@@ -73,26 +73,23 @@ def construct_global_ctx(self):
     def db_updater(self):
         rc = self.rc
         name = HumanName(rc.name)
-        frag = fragment_retrieval(self.gtx['contacts'], ["_id", "aka", "name"], rc.name)
-        ids = set(i.get('_id') for i in frag)
-        sorted_ids = sorted(ids)
-        index = 1
-        all_names = {}
-        for i in sorted_ids:
-            currentid = rc.client.find_one(rc.database, rc.coll, {'_id': i})
-            index += 1
-            all_names[str(index)] = {"_id":i, "name":currentid.get("name")}
+        sorted_ids = sorted(set(i.get('_id') for i in fragment_retrieval(self.gtx['contacts'],
+                                            ["_id", "aka", "name"], rc.name)))
+        index = list(range(len(sorted_ids)))
         if not rc.number:
             print("Please choose from one of the following to update/add:")
-            print(f"1. {rc.name} as a new contact")
-            for k, v in all_names.items():
-                print(f"{k}. {v['name']}    id: {v['_id']} ")
+            print(f"{0}. {rc.name} as a new contact")
+            print(*[f"{i+1}. {rc.client.find_one(rc.database, rc.coll, {'_id': j})['name']}    id: {j}\n"
```

Comment:
I guess what I was trying to avoid was multiple calls to the db which are slow.  I think the fragment_retrieval gives you everything you need so we can avoid the second db call

Reply:
just to expand on that...you will have to modify the code I gave you to do it in two steps, sthg like
```
found_contacts = fragment_retrieval....
sorted_ids = list(set([i.get("_id") for i in found_contacts])).sort()
```
higher up, but then everything you need for the print you can get directly from ``found_contacts``

Reply:
done!